cli/singleexecutor.cpp

```diff
@@ -49,9 +49,9 @@ unsigned int SingleExecutor::check()
     std::size_t processedsize = 0;
     unsigned int c = 0;
 
-    for (std::list<FileWithDetails>::const_iterator i = mFiles.cbegin(); i != mFiles.cend(); ++i) {
```

Comment:
We have not changed such loops yet since they are in a mutable scope and you cannot fully enforce the intent of constness in a range loop until C++17 which introduces `std::as_const` (I have been trying to add a helper for that).

That seems pedantic but since changing the code does not actually change anything it doesn't matter which style you use.

Reply:
Coincidentally I finally ran into some code which highlights the issue - see #6653.